src/stores/constraints.ts

```diff
@@ -227,8 +210,24 @@ export const cachedConstraintsStatus: Readable<Status | null> = derived(
   ([$relevantConstraintRuns, $constraintPlanSpecsMap]) => {
     return $relevantConstraintRuns.reduce(
       (status: Status | null, constraintRun: ConstraintRun) => {
+        const constraintPlanSpec =
+          $constraintPlanSpecsMap[constraintRun.constraint_id]?.[constraintRun.constraint_invocation_id];
+        let constraintPlanSpecRevision: number | null = constraintPlanSpec.constraint_revision ?? null;
+        if (constraintPlanSpec.constraint_revision === null) {
+          const constraintVersions = constraintPlanSpec.constraint_metadata?.versions ?? [];
+          constraintPlanSpecRevision = constraintVersions[constraintVersions.length - 1].revision;
+        }
+
+        if (
+          constraintPlanSpec &&
+          (constraintRun.constraint_revision !== constraintPlanSpecRevision ||
+            JSON.stringify(constraintRun.arguments) !== JSON.stringify(constraintPlanSpec.arguments))
+        ) {
+          return Status.Modified;
+        }
+
         if (constraintRun.results.violations?.length) {
-          return Status.Failed;
+          return Status.Complete;
```

Comment:
Why change this to Complete instead of Failed? I thought the overall status should be the worst outcome of any of the sub status items (did the constraints run, did any of them have violations). Or did folks decide to change this approach? If the latter, we should probably update how we communicate status of simulation and other similar status bound actions.
<img width="1169" alt="image" src="https://github.com/user-attachments/assets/c672f652-a496-4eae-baa6-0f6a59a07aba" />

Reply:
Also the "2 constraints have violations" in the nav should probably have a failure icon next to it, may all come from the same logic though.

Reply:
Should the status icon in the constraints panel also show the worst case?
![image](https://github.com/user-attachments/assets/f2dbba5b-7f09-4aea-bd4c-cafef0ab0c9b)

Reply:
Fixed! Hopefully good to go!